documentation/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,383 @@
+  - [Excessive Recompilation](#excessive-recompilation)
+- [Accuracy Debugging](#accuracy-debugging)
+- [File an Issue](#file-an-issue)
+
+# Introduction
+
+TorchDynamo is still in active development, and many of the reasons for graph breaks and excessive recompilation will be fixed with upcoming support for [tracing dynamic tensor shapes](https://docs.google.com/document/d/1QJB-GOnbv-9PygGlOMXwiO9K6vVNm8sNg_olixJ9koc/edit?usp=sharing), more careful choices for guards and better tuned heurstics.
+
+In the mean time, you may need to diagnose a particular issue and determine if it is easy to work around with a change to your model, or file an issue for support.
+
+We're also actively developing debug tools, profilers, and improving our errors/warnings.  Please give us feedback if you have an issue with this infra, or an idea for an improvement. Below is a table of the available tools and their typical usage. For additional help see the next section.
+| Tool                                | Purpose                                                                                                                      | Usage                                                                        |
+| ----------------------------------- | ---------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------- |
+| Debug logging                       | View detailed steps of compilation (print every instruction traced)                                                          | `torchdynamo.config = logging.DEBUG` and `torchdynamo.config.verbose = True` |
+| Minifier for any backend            | Find smallest subgraph which reproduces errors for any backend                                                               | set environmet variable TORCHDYNAMO_REPRO_AFTER="torchdynamo"                |
+| Minifier for TorchInductor          | If the error is known to occur after AOTAutograd find smallest subgraph wich reproduces errors during TorchInductor lowering | set environment variable TORCHDYNAMO_REPRO_AFTER="aot"                       |
+| `torchdynamo.explain`               | Find graph breaks and display reasoning for them                                                                             | `torchdynamo.explain(fn, *inputs)`                                           |
+| Record/Replay (in progress)         | Record and replay frames which to reproduce errors during graph capture                                                      | `torchdynamo.config.replay_record_enabled = True`                            |
+| TorchDynamo function name filtering | only compile functions with the given name to reduce noise when debugging an issue                                           | set environment variable TORCHDYNAMO_DEBUG_FUNCTION=\<name\>                 |
```

Comment:
We also have `TORCHINDUCTOR_TRACE=1` https://github.com/pytorch/torchdynamo/pull/1071

Reply:
Done. Added section with your explanation and examples from your PR as well.